src/codesyncd/handlers/tab_handler.ts

```diff
@@ -0,0 +1,85 @@
+import fs from "fs";
+import os from "os";
+import path from "path";
+
+import { ITabYML, ITabFile, ITabToSend } from "../../interface";
+import {generateSettings} from "../../settings";
+import {readYML} from "../../utils/common";
+import {CodeSyncLogger} from "../../logger";
+import {pathUtils} from "../../utils/path_utils";
+import {initUtils} from "../../init/utils";
+import {VSCODE} from "../../constants";
+import { removeTabFile } from "../../utils/tab_utils";
+
+export class TabHandler {
+    accessToken: string;
+	tabData: ITabYML;
+	tabFilePath: string;
+	repoPath: string;
+
+	repo_id: number;
+	created_at: string;
+	source: string;
+	file_name: string;
+	tabs: ITabFile[];
+
+	configJSON: any;
+    configRepo: any;
+
+	constructor(
+		tabData: ITabYML,
+		tabFilePath: string | null = null,
+		repoPath: string,
+		accessToken: string
+	) {
+		this.accessToken = accessToken;
+		this.tabData = tabData;
+		this.repo_id = tabData.repo_id;
+		this.created_at = tabData.created_at;
+		this.source = tabData.source;
+		this.file_name = tabData.file_name;
+		this.tabs = tabData.tabs;
+		this.repoPath = repoPath
+		if (!tabFilePath) return;
```

Comment:
what is tabFilePath and why will it be null?

Reply:
This will be the path of a tab,

I set its type as `string || null` as in the case of `tabs_handler.run()`, we need to call `tab_handler.createTabToSend()`. And at that point in type we won't have access to paths of tabs I believe. 

Path is required to remove tab files, and at that point of time we should have access to path, so we can instantiate constructor with actual `tabFilePath` value

Reply:
Do you mean TabYML file from buffer?
> This will be the path of a tab,